### DISCUSSION TITLE
about log/return request traceid(skywalking tid) in APISIX

### DISCUSSION THREAD

**Discussion by ambluse** [apache/apisix#5358] [General]
can we use "response rewrite" plugin to  set  the traceid（skywalking tid）header?  if not ，where can get this traceid and log it (like APISIX variables? )， any ideas？

**Comment by juzhiyuan** [comment]
Hi @tzssangglass, could you please have a look? 😄

**Reply by tzssangglass** [reply]
I'm not very familiar, @dmsolr do you know?

**Comment by dmsolr** [comment]
Yes, you can. 
You could get the trace context header from request headers. and then get the trace id from there.
Refer to [Skywalking-Cross-Process-Propagation-Headers-Protocol-v3](https://github.com/apache/skywalking/blob/master/docs/en/protocols/Skywalking-Cross-Process-Propagation-Headers-Protocol-v3.md).

**Reply by tzssangglass** [reply]
@ambluse If this reply solves your problem, pls mark it as answer.
